internal/transport/transport.go

```diff
@@ -204,7 +204,7 @@ func NewTransport(nhConfig config.NodeHostConfig,
 	dir server.SnapshotDirFunc, sysEvents ITransportEvent,
 	fs vfs.IFS) (*Transport, error) {
 	sourceID := nhConfig.RaftAddress
-	if nhConfig.AddressByNodeHostID {
+	if nhConfig.AddressByNodeHostID || nhConfig.Expert.NodeRegistryFactory != nil {
```

Comment:
This check is used in multiple places, I guess it should be replaced by a new method like `nhConfig.NodeRegistryEnabled() bool`. 

`nhConfig.AddressByNodeHostID` seems no longer to be the best name here, as I *guess* a typical custom node registry will still rely on such NodeHostID values. Maybe rename `nhConfig.AddressByNodeHostID` to `nhConfig.DefaultNodeRegistryEnabled`.

Reply:
First part is Done.

Second part also done -- but this change blows up this PR a bit. I put it in a separate commit so it's easy to undo if we don't want to do it in this PR.